src/components/Rocket.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import { addReservation as addRes, removeReservation as rmRes } from '../redux/rockets/rockets';
+import '../css/Rocket.css';
+
+const Rocket = ({ rocket }) => {
+  const dispatch = useDispatch();
+
+  const {
+    description, image, name, id,
+  } = rocket;
+
+  const isReserved = (rocket) => {
+    if (rocket.reserved) return true;
+    return false;
+  };
+
+  const handleDispatch = () => (rocket.reserved ? dispatch(rmRes(id)) : dispatch(addRes(id)));
+
+  return (
+    <div className="flex items-center w-full gap-4 rocket-container">
+      <img className="rocket-image" src={image} alt={name} />
+      <div>
+        <h2 className="text-xl font-semibold">{name}</h2>
+        <p className="text-md">
+          {isReserved(rocket) && (
+            <span className="bg-blue-400 text-white text-xs p-1 text-center mr-2 rounded-md">Reserved</span>
+          )}
+          {description}
+        </p>
+        <button
+          className={isReserved(rocket) ? 'text-sm text-blue-500 bg-white border-1 border-blue-500 rounded-md p-2' : 'text-sm border-1 border-blue-500 text-white bg-blue-500 rounded-md p-2'}
+          onClick={handleDispatch}
+          type="button"
+        >
+          {isReserved(rocket) ? 'Cancel Reservation' : 'Reserve Rocket'}
```

Comment:
According to the project requirements:

![image](https://user-images.githubusercontent.com/57421823/162113536-4fee3642-4cd9-4baf-b87f-508f39817e25.png)

Therefore:

- [ ] Please, kindly use the React conditional rendering syntax, instead of using ternary operators.

## Explanation

This React rendering syntax, is commonplace in the workplace, and we need to practice that.

## Hints

Everything after the `&&` means "if the left part is true, do the things in the right part".

So in this case:

```JAVASCRIPT
{rocket.reserved && ( 
    // render Cancel button
)}
```

It means that if the rocket is reserved is TRUE, then it will do the thing on the right (...) after the `&&`.

You can build if, then, else schemas by using `!!`:

```JAVASCRIPT
{!!rocket.reserved && ( 
    // don't render the cancel button
)}
```

_Note: This is just an example to give you an idea of how to make the solution work.  You should use this as a mini-tutorial and implement what you need, according to your own project needs.  This is just the result of my own personal experience and effort, and it is by no means to be taken as an "official" solution._

I hope this is helpful to you! 💪